### ISSUE TITLE
Non intuitive error when placing exceptions in the wrong order

### ISSUE THREAD

**Issue by Enchufadoo** [nim-lang/Nim#9382]
I'm following the tutorial and I noticed that when you try to catch a general exception before a particular exception the IDE (vscode) doesn't detect the error and the error output is kind of weird

```
from strutils import parseInt
var
  f: File
if open(f, "numbers.txt"):
  try:
    let a = readLine(f)
    let b = readLine(f)
    echo "sum: ", parseInt(a) + parseInt(b)
  except OverflowError:
    echo "overflow!"
  except ValueError:
    echo "could not convert string to integer"
  except:
    echo "Unknown exception!"
    # reraise the unknown exception:
    raise
  except IOError:
    echo "IO error!"
  finally:
    close(f)
```

Produces
```
Error: execution of an external compiler program 'gcc -c  -w  -I/home/ariel/.choosenim/toolchains/nim-0.19.0/lib -o /home/ariel/.cache/nim/veinticuatro_d/veinticuatro.c.o /home/ariel/.cache/nim/veinticuatro_d/veinticuatro.c' failed with exit code: 1

/home/ariel/.cache/nim/veinticuatro_d/veinticuatro.c: In function ‘NimMainModule’:
/home/ariel/.cache/nim/veinticuatro_d/veinticuatro.c:607:4: error: ‘else’ without a previous ‘if’
    else    if (isObj(getCurrentException()->Sup.m_type, (&NTI_iLZrPn9anoh9ad1MmO0RczFw_))) {
    ^~~~
```
Thanks.

**Comment by skilchen** [comment]
In Nim 0.19.1 you get:
```
Error: Only one general except clause is allowed after more specific exceptions
```

**Comment by mratsim** [comment]
Fixed 7 days ago by #9191:

https://github.com/nim-lang/Nim/blob/0c04da6f2644f97ff86d541e0c95dd92c7fb7753/compiler/semstmts.nim#L258-L261
